app.rb

```diff
@@ -0,0 +1,149 @@
+require './book'
+require './classroom'
+require './base_decorator'
+require './person'
+require './rental'
+require './student'
+require './teacher'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def start
+    selecting
+  end
+
+  def books_list
+    puts 'No books available!!' if @books.empty?
+    puts "There are #{@books.length} books available"
+
+    @books.each_with_index do |book, index|
+      puts "#{index + 1}) Title: \"#{book.title}\" | Author: #{book.author}"
+    end
+  end
+
+  def peoples_list
+    puts 'No added people!! Please Add new person.' if @people.empty?
+    puts "#{@people.length} peoples added."
+
+    @people.each_with_index do |person, index|
+      puts "#{index + 1})[#{person.class}] Name: #{person.name} | Age: #{person.age} | ID: #{person.id}"
+    end
+  end
+
+  def create_person
+    puts 'Would like to create student (1) or teacher (2)'
+    choice = gets.chomp
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Age:'
+    age = gets.chomp.to_i
+
+    case choice
+    when '1'
+      create_student(age: age, name: name)
+    when '2'
+      create_teacher(age: age, name: name)
+    else
+      puts 'Invalid. Please try again!'
+    end
+  end
+
+  def create_student(age:, name:)
+    puts 'Parent permission [Y/N]'
+    choice = gets.chomp.downcase
+
+    case choice
+    when 'y'
+      student = Student.new(age: age, name: name, parent_permission: true)
+    when 'n'
+      student = Student.new(age: age, name: name, parent_permission: false)
+    else
+      puts 'Invalid. Try again'
+    end
+    @people << student
+    puts 'Student created successfully!!'
+  end
+
+  def create_teacher(age:, name:)
+    puts 'Please add your specialization:'
+    specialization = gets.chomp
+    teacher = Teacher.new(age: age, name: name, specialization: specialization)
+    @people << teacher
+    puts 'Teacher created successfully!!'
+  end

```

Comment:
- [ ] The app crashes with the attached error message when I tried creating a teacher. Kindly fix this to make your app work as expected.
<img width="483" alt="chrome_4Sipp1CKoH" src="https://user-images.githubusercontent.com/81468675/175521704-e5667706-d6b0-462d-b556-65e33d3bc502.png">

Reply:
**Well done fix the error.** 🥇